src/backend-proxy.js

```diff
@@ -47,10 +48,24 @@ function tryReadData(options, backendResponse, request, next) {
 
 function createHandler({request, response, next, options, backendHttpOptions}) {
 	return backendResponse => {
+		let interceptErrors = false;
```

Comment:
Just a thought:

You could do the `if/else` check outside this function and assign it to a function that you can call each time?

```
let interceptErrors;

if (typeof options.interceptErrors === 'function') {
      interceptErrors = options.interceptErrors;
} else if (options.interceptErrors === true) {
      interceptErrors = () => true;
} else {
      interceptErrors = () => false;
}


/// then in the handler
if (backendResponse.statusCode >= 400 && backendResponse.statusCode <= 599 && interceptErrors(backendResponse)) {
     return next({statusCode: backendResponse.statusCode});
}

Reply:
Good call! Will do